app/services/scenes/scene-folder.ts

```diff
@@ -108,12 +106,19 @@ export class SceneItemFolder extends SceneItemNode implements ISceneItemFolderAp
     });
   }
 
-  getNestedNodes(): TSceneNode[] {
+  getNestedNodes(traversedNodesIds: string[] = []): TSceneNode[] {
+    traversedNodesIds = [].concat(traversedNodesIds);
     const nodes: TSceneNode[] = [];
     this.getNodes().forEach(node => {
+      if (traversedNodesIds.includes(node.id)) {
+        // TODO: find the use-case that causes loops in folders structure
+        console.error(`Loop in folders structure detected', ${this.name} -> ${node.name}`);
+        return;
+      }
```

Comment:
Lol.  Did our tree structure accidentally become a directed graph?

![2e496w](https://user-images.githubusercontent.com/3180181/42774638-0f3b8cba-88e7-11e8-8b32-ced5d67ea9ec.jpg)

Reply:
I think so. That explains the infinity recursion in the so simple function.